apollo-router/src/http_subgraph.rs

```diff
@@ -15,14 +16,14 @@ type BytesStream = Pin<
 #[derivative(Debug)]
 pub struct HttpSubgraphFetcher {
     service: String,
-    url: String,
+    url: Url,
     #[derivative(Debug = "ignore")]
     http_client: reqwest_middleware::ClientWithMiddleware,
 }
 
 impl HttpSubgraphFetcher {
     /// Construct a new http subgraph fetcher that will fetch from the supplied URL.
-    pub fn new(service: String, url: String) -> Self {
+    pub fn new(service: String, url: Url) -> Self {
```

Comment:
:yarn: The public API will be nicer to use this way. (And it will simplify your tests a bit)

```suggestion
    pub fn new(service: impl Into<String>, url: Url) -> Self {
```

Reply:
fixed in 6c0a2c8